lib/phctl.rb

```diff
@@ -203,6 +180,72 @@ def weeding_decision(organization)
     end
   end
 
+  class Workflow < JobCommand
+    class_option :records_per_job, type: :numeric, default: Settings.mapreduce.records_per_job
+    class_option :test_mode, type: :boolean, default: false
+
+    no_commands do
+      def component(...)
+        WorkflowComponent.new(...)
+      end
+
+      def run_workflow(components)
+        params = base_params.merge(components: components)
+        if parent_options[:inline]
+          Jobs::MapReduceWorkflow.perform_now(**params)
+        else
+          Jobs::MapReduceWorkflow.perform_async(**params)
+        end
+      end
+
+      def base_params
+        {
+          records_per_job: options[:records_per_job],
+          test_mode: options[:test_mode]
+        }
+      end
+    end
+
+    desc "costreport --ht-item-count NUM --ht-item-pd-count NUM (--chunk-size SIZE)", "Dump records from solr, split into chunks of chunk-size records, generate frequency tables for each chunk, sum the resulting frequency tables, and generate a cost report based on that table."
+    option :ht_item_count, type: :numeric
+    option :ht_item_pd_count, type: :numeric
+    def costreport_workflow
+      components = {
```

Comment:
I spent a fair amount of time trying to come up with an option for instantiating a workflow that could:
* have the same interface in phctl / via sidekiq as when directly running in testing
* put the responsibility for marshalling/unmarshalling parameters in `SidekiqJobs`
* not involve confusing stuff in `Workflows::MapReduce`

I don't know if I completely succeeded, but hopefully this is a reasonable compromise among those ends.